lib/_http_outgoing.js

```diff
@@ -426,6 +430,37 @@ OutgoingMessage.prototype.getHeader = function getHeader(name) {
 };
 
 
+// Returns an array of the names of the current outgoing headers.
+OutgoingMessage.prototype.getHeaderNames = function getHeaderNames() {
+  return (this._headers ? Object.keys(this._headers) : []);
+};
+
+
+// Returns a shallow copy of the current outgoing headers.
```

Comment:
shallow can mean "copied as-is, so don't touch the values, you will mess http up" or "the values are copied as-is, so they can be modified". "shallow" describes what the code does, not how the API is allowed to be used.